ops/pebble.py

```diff
@@ -1565,11 +1793,12 @@ def _wait_change(self, change_id: ChangeID, timeout: float = None) -> Change:
 
         return Change.from_dict(resp['result'])
 
-    def _wait_change_using_polling(self, change_id, timeout, delay):
+    def _wait_change_using_polling(self, change_id: ChangeID, timeout: Optional[float],
+                                   delay: float):
         """Wait for a change to be ready by polling the get-change API."""
         deadline = time.time() + timeout if timeout is not None else None
 
-        while timeout is None or time.time() < deadline:
+        while timeout is None or time.time() < deadline:  # type: ignore
```

Comment:
This is a bit confusing.  pyright is too dumb to figure out that deadline and timeout are both either None or not None together.  But this also basically means that deadline could just have `if timeout is not None else 0` and then we don't need any type annotations anywhere.

Reply:
True, true, again I tried to refrain from introducing innocent-seeming refactors as part of this PR, fearing to add subtle bugs. In this specific case it does look very innocent, so... @benhoyt can you help confirm?